ChronoVisor/src/VisorClientPortal.cpp

```diff
@@ -253,18 +248,12 @@ int chronolog::VisorClientPortal::ReleaseStory(chl::ClientId const &client_id, s
     { return CL_ERR_NOT_AUTHORIZED; }
 
     StoryId story_id(0);
-    bool notify_keepers = false;
-    auto return_code = chronicleMetaDirectory.release_story(client_id, chronicle_name, story_name, story_id
-                                                            , notify_keepers);
+    auto return_code = chronicleMetaDirectory.release_story(client_id, chronicle_name, story_name, story_id);
     if(chronolog::CL_SUCCESS != return_code)
     { return return_code; }
 
-    if(notify_keepers && theKeeperRegistry->is_running())
-    {
-        std::vector <chronolog::KeeperIdCard> recording_keepers;
-        theKeeperRegistry->notifyKeepersOfStoryRecordingStop(theKeeperRegistry->getActiveKeepers(recording_keepers)
-                                                             , story_id);
-    }
+    theKeeperRegistry->notifyRecordingGroupOfStoryRecordingStop(story_id);
```

Comment:
Should we notify Keepers and Graphers first before we update metadata?

Reply:
I don't think the order of operations matters here. Even if notification fails for some reason the story recording still stops and the metadata records  should reflect this fact

Reply:
> Even if notification fails for some reason the story recording still stops

I don't get it here. If those two RPCs to Keepers and Graphers fail, Keepers and Graphers will not receive notification to free the data structure for this Story. Will this affect how Keepers seal StoryChunk and extract to Grapher?

Reply:
StoryChunks are extracted from the StoryPipeline  and moved to the ExtractionQueue based on time: as soon as acception window has passed the chunk is moved to ExtractionQueue. 

notification message send out to Keeper and Grapher processes does not "seal" StoryChunks in any way different from before the notify. The notification message tells the keeper/grapher process that the StoryPipeline for this particular story can be dismantled after it's drained of the remaining StoryChunks